### PR TITLE
Metadata Typing

### PR DESCRIPTION
**Pull request by ric-evans**
- added `schema/types.py` containing metadata type hints
     + this can be used by clients (aka the indexer)
     + this could allow `validation.py`  to validate metadata using introspection (future)
- moved `validation.py` to `schema/validation.py`

### CODE REVIEW

File: file_catalog/schema/types.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    # i3 File:
+    data_type: Optional[str]
+    processing_level: Optional[str]
+    content_status: str
+
+    # /data/exp/* i3 File:
+    software: Optional[List[SoftwareEntry]]
+    run: Run
+    offline_processing_metadata: OfflineProcessingMetadata
```
Comment (dsschult) [inline file_catalog/schema/types.py:178]: Are we really using such a long name?

**Review by dsschult — COMMENTED** [review]
Type information generally looks good.  I guess one question is how to fit the rest of the columns in the metadata doc.

File: file_catalog/schema/types.py (hunk 1 above)
Comment (ric-evans) [reply file_catalog/schema/types.py:178]: Yup, I think this is the longest name

**Comment by ric-evans** [comment]
> Type information generally looks good. I guess one question is how to fit the rest of the columns in the metadata doc.

We can put descriptions as an inline comment. For whether the key is required, that would need to be an independent dict.

**Comment by ric-evans** [comment]
> > Type information generally looks good. I guess one question is how to fit the rest of the columns in the metadata doc.
> 
> We can put each description as an inline comment. For whether the key is required, that would need to be an independent dict.

We could really go all out and duplicate/migrate the Google Doc into a dict/yaml. But this is beyond the scope of this PR

**Comment by dsschult** [comment]
> > > Type information generally looks good. I guess one question is how to fit the rest of the columns in the metadata doc.
> > 
> > 
> > We can put each description as an inline comment. For whether the key is required, that would need to be an independent dict.
> 
> We could really go all out and duplicate/migrate the Google Doc into a dict/yaml. But this is beyond the scope of this PR

Fair enough.

**Comment by ric-evans** [comment]
> > > > Type information generally looks good. I guess one question is how to fit the rest of the columns in the metadata doc.
> > > 
> > > 
> > > We can put each description as an inline comment. For whether the key is required, that would need to be an independent dict.
> > 
> > 
> > We could really go all out and duplicate/migrate the Google Doc into a dict/yaml. But this is beyond the scope of this PR
> 
> Fair enough.

https://github.com/WIPACrepo/file_catalog/issues/81

**Pull request merged**